File: ruby/rectangle.rb

require "./shape.rb"
class Rectangle < Shape
    attr_accessor :height, :width
    @@valid = true
    # constructor
    def initialize(height, width)
        super()
        if height.to_i < 0 || width.to_i < 0 then
            @@valid = false
            @@height = 0
            @@width = 0
        else
            @@height = height.to_i
            @@width = width.to_i
        end
    end

    # Overriden perimeter method 
    # 
    def perimeter
       return 2 * (@@height + @@width)
    end 

    # area of the circle
    def area
        return @@height * @@width
    end

    def isValid()
        return @@valid
    end

end

File: aspectj/a3_40176765/src/a3_40176765/Rectangle.java
package a3_40176765;

public class Rectangle {
	
	private double width,height;
	
	public Rectangle(double width, double height) {
		this.width = width;
		this.height = height;
	}
	
	public double getPerimeter() {
		return 2 * (this.width + this.height);
	}
	
	public double getArea() {
		return this.height*this.width;
	}
	
}

